### ISSUE TITLE
when guicursor set to blinking, the cursor starts in 'off' mode before blinking

### ISSUE THREAD

**Issue by Frydac** [Kethku/neovide#577]
EDIT: you can ignore this initial post, please look at the 3rd message: https://github.com/Kethku/neovide/issues/577#issuecomment-823240031

Hi,

When I hold e.g. `l` and let the keyrepeat kick in, the cursor is not visible during the hold time.
If I try this in neovim shell or gvim/vim the cursor is visible.

I know I shouldn't be holding movement keys.. but it happens from time to time :)
(I haven't set any of the neovide specific settings)

**Comment by Frydac** [comment]
Oops, my bad.
I did have a custom guicursor option set when loading nvim I forgot about, which causes this issue:
```    
set guicursor=a:blinkwait100-blinkon300-blinkoff300
```
It is still somewhat weird that it doesn't happen in neovim.
I'll try without the blinking, see if I can get used to it :)

**Comment by Frydac** [comment]
I experimented a bit with it, and it seems there might be some difference behavior that might indicate a bug:

when I for example set:
```
set guicursor=a:blinkwait2000-blinkon300-blinkoff300
```
It should wait with blinking for 2000ms
when I use nvim after moving the cursor it starts with the cursor 'on' for 2s and then starts blinking
in neovide after moving the cursor it seems the cursor starts with the cursor in 'off' for 2s before it starts blinking.

**Comment by Kethku** [comment]
Yeah thats a bug. The actual logic is found here: https://github.com/Kethku/neovide/blob/main/src/renderer/cursor_renderer/blink.rs

**Comment by Kethku** [comment]
fixed by #668
